teku/src/main/java/tech/pegasys/teku/cli/BeaconNodeCommand.java

```diff
@@ -216,15 +222,48 @@ public int parse(final String[] args) {
 
     final Optional<File> configFile = getConfigFileFromCliOrEnv(configFileCommand);
 
-    // final pass
+    // second pass to collect matched params and enrich with additional (environment and\or config
+    // file)
     final CommandLine commandLine = getCommandLine();
     commandLine.setCaseInsensitiveEnumValuesAllowed(true);
+
+    final ParseResult parseResult;
+    try {
+      parseResult = commandLine.parseArgs(args);
+    } catch (ParameterException e) {
+      return handleParseException(e, args);
+    }
+
+    // switch to last subcommand if needed
+    final ParseResult maybeSubCommandParseResult =
```

Comment:
nit: The maybe in the name here keeps making me think it's an `Optional`.  Maybe we should just reuse the `parseResult` field here and reassign it? Or just find a name without the maybe.